IBX-7076: Customizable dashboard described in User Documentation (#246)
ibexa/documentation-user: docs/getting_started/discover_ui.md

# User interface

## Dashboard

The first screen that you see after logging in is [Dashboard](dashboard.md).

> It is default dashboard and contains selected blocks.

Reviewer: ```suggestion
It is the default dashboard and contains selected blocks.
```

![Dashboard](dashboard.png "Dashboard")

> You can also customize dashboard. For more information, see [Work with dashboard](work_with_dashboard.md).

Reviewer: ```suggestion
You can customize the dashboard by changing the blocks and their layout.
For more information, see [Work with dashboard](work_with_dashboard.md).
```

# User interface

## Dashboard

The first screen that you see after logging in is [Dashboard](dashboard.md).
It is the default dashboard and contains selected blocks.

![Dashboard](dashboard.png "Dashboard")

You can customize the dashboard by changing the blocks and the layout.
For more information, see [Work with dashboard](work_with_dashboard.md).

> ## Menu

Author: "Menu" it's not in the scope of this PR. Will be updated in separated PR.

Available filters are: User(s), Activity type (create, update, publish, delete, hide, show) and Activity area (product, content).

- **Ibexa news** - a list of recent blog posts or articles published at `ibexa.co` blog.
It includes title, image, timestamp, and link to article details. List can containt minimum 1 and maximum 10 links.

- **Review queue** - a list of Content items which user or User group can review.

- **My content** - a list of Content items created by the user currently logged in.

- **Common content** - a list of Content items created by all the organization that the user belongs to. You can select any of these items and view them in detail or start [editing them](create_edit_content_items.md) from this point.
You can also [reschedule or cancel planned publications](../content_management/schedule_publishing.md#reschedule_or_cancel_publications) using the dashboard.

![Dashboard](img/dashboard.png "Dashboard")

From any page you can always return to the **Dashboard** screen by clicking the logo in the upper left corner.

![Dashboard icon](img/dashboard_icon.png)

> ## Customizable dashboard

Author: This section describes customizable dashboard. Inside of this section, there is "Customize dashboard" subsection with the explanation, how to do it.

You can customize the dashboard depending on your needs.
Customized dashboard previews a set of widgets selected by the user.

It provides quick overview of these areas, that are commonly used, excluding those that are not used, or used rarely.
This solution improves productivity, allows to enhance the default dashboard with additional widgets, and helps to make better business decisions based on data.

By highlighting important information, users can work more efficiently with a clear view of tasks and deadlines.

> Customization includes also choosing one of predefined layouts and adding custom ones to use them in dashboards.

Reviewer: How about changing the wording of this section and moving the remaining procedure to a separate article:

"Customization allows you to quickly visit the areas that are used more often and exclude those less important:
- Selected widgets boost productivity by highlighting vital data and helping in decision making
- Predefined and custom layouts improve visibility for efficient task and deadline management"

For more information, see "Work with dashboard".

|Structure view|![Structure view](img/dashboard_structure_view.png)|Access Structure view toolbar.|
|Elements menu|![Elements menu](img/dashboard_elements.png)|Access Elements toolbar.|
|View switch|![View switch](img/dashboard_switch_toolbar.png)|Move Elements / Structure toolbar to the other side of the screen.|

Dashboard Builder has two main toolbars that you can use while creating a dashboard:

- **Elements** - consists of all available blocks that you can use by dragging them and dropping on a drop zone.

![Elements](img/dashboard_elements_toolbar.png)

- **Structure view** - shows a structure of your dashboard, including its division into zones and the blocks that they contain.

![Structure view](img/dashboard_structure_view_toolbar.png)

### Choose layout

For new dashboard you can choose layout which defines the available zones.
Applying a layout divides the dashboard into the defined zones. The zones are placeholders for blocks.

To do it, click the **Switch layout** icon on the left side of the upper toolbar, then choose one from available layouts, and click **Submit**.

![Switch layout](img/dashboard_switch_layout_window.png)

#### Add blocks

> In Dashboard Builder you can access a menu of **Elements** — it includes a set of blocks of content that you can add to the zones of the dashboard.

Reviewer: ```suggestion
In Dashboard Builder you can access a menu of **Elements** which includes a set of blocks of content that you can add to the zones of the dashboard.
```

> You can choose from available blocks:
> 
> - Recent orders
> - Orders by status
> - Ibexa news
> - Quick actions
> - My content
> - Common content
> - Review queue
> - Recent activity
> - Products with lowest stock
> - Products by category
> - Top 10 clicked items

Reviewer: ```suggestion
```

> 
> !!! Tip
> 
>     For detailed description of available blocks, see [Dashboard block reference](dashboard_block_reference.md).

Reviewer: ```suggestion
For a list of available blocks and their detailed description, see [Dashboard block reference](dashboard_block_reference.md).
```

Add a block by dragging it from the menu to an empty place on a zone.
Do not worry about placing blocks in the proper place from the start.
You can reorder them at any time by dragging and dropping them in the desired location.
You can reorder blocks in a few ways:

- drag and drop block in the desired location on a drop zone
- use **Move up block** or **Move down block** option from the block toolbar
- access Structure view and use 'Move up' and 'Move down' function in the settings of the block

> When you add a new block to the drop zone, drop it in the blue highlighted area. Before you drop it, a bold line appears  - it helps you see the position of the newly added block in relation to other, already added blocks.

Reviewer: ```suggestion
When you add a new block to the drop zone, drop it in the blue highlighted area.
Before you drop it, a bold line appears that it helps you see the position of the newly added block in relation to existing blocks.
```

![Drop zone line](img/dashboard_blue_line.png)

> When you add a block by dragging it from Elements menu into the drop zone,
> the block settings panel open immediately where you can configure all block properties.

Reviewer: ```suggestion
When you add a block by dragging it from Elements menu into the drop zone, the block settings panel opens immediately, and you can configure all block properties.
```

![Block properties](img/dashboard_block_properties.png)

> This is a default behavior. You can globally turn off automatic opening of the block settings panel in the user settings.

Reviewer: ```suggestion
This is a default behavior. 
You can globally disable the automatic automatic opening of the block settings panel in the user settings.
```

First, access your user settings on the right side of the top bar:

![User settings](img/user_settings.png "User settings")

Then, go to **Preferences** tab, **Edit** section.

> Here, you can find `Automatically open block settings in builder` setting, which, by default, is set up to `enabled` value.

Reviewer: ```suggestion
Here, you can find `Automatically open block settings in builder` setting, which is enabled by default.
```

To change this behavior, click on **Edit** icon, find the setting, and change its value to `disabled`.

![Blocks settings](img/user_settings_blocks.png "User settings - blocks settings")

#### Work with blocks

Each block has its special properties.

> You can access them by placing the cursor on the added block and clicking the **Block settings** icon.

Reviewer: ```suggestion
You can access them by placing the cursor on a block in the drop zone and clicking the **Block settings** icon.
```

![Block settings - Dashboard Builder](img/block_settings_builder.png "Block settings - Dashboard Builder")

Settings available for blocks are divided into two tabs: **Properties** and **Design**.

> The settings available on the **Properties** tab are tailored to the block's content.

Reviewer: ```suggestion
The settings available on the **Properties** tab depend on the block's content.
```

For a description of these settings, see [Dashboard block reference](dashboard_block_reference.md).

To save changes, click **Save and close**.

#### Edit dashboard

At any time you can edit an active customized dashboard.

> To do it, click an icon in the top right corner, and then click **Edit My dashboard**.

Reviewer: ```suggestion
To do it, click the three dots icon in the top right corner, and then click **Edit My dashboard**.
```
As far as I can see, the label says "Edit >>dashboard name<< dashboard" and changes dynamically.
If it is so, I would say "... click "Edit My dashboard", where "My" can be replaced by the dashboard name that you defined in  Dashboard Builder."

However, I just created a dashboard on nightly and after switching to it I did not see this option. Is it a bug or there is a permission of some sort required?

![Edit dashboard](img/edit_active_dashboard.png "Edit dashboard")

This action opens Dashboard Builder that allows you to edit an active customized dashboard and publish its updated version.
You can see all the implemented changes after reloading starting screen.

> #### Select active dashboard

Reviewer: ```suggestion
### Select active dashboard
```
Selecting a dashboard is not related to customizing, I'd move it one level up in the outline, on the same level as "Customize..."

> When you create new customized dashboard, you can choose, which dashboard you want to set as an active.

Reviewer: ```suggestion
When multiple dashboards exist in the system, including your customized ones, you can choose, which dashboard is set as active.
```
I imagine the product could come with several predefined dashboards, not only the default one, and the custom one created by the user.

Author: No, for now it is one default and one predefined dashboard possible.

Reviewer 2: There are only two dashboards in the system -> default one and a customizable copy, so multiple word here isn't correct.

To do it, access your user settings on the right side of the top bar.
Then, go to **Preferences** tab, **Dashboard** section, and click **Edit**.

Select the dashboard, that you want to set as active, and click **Save**.

![Select active dashboard](img/select_active_dashboard.png "Select active dashboard")

> #### Permissions

Reviewer: This is a prerequisite and should come right after the introductory section of the "Work with dashboards" article.

> In order to be able to customize dashboard, you need to have `dashboard/cutomize` policy.

Reviewer: ```suggestion
To be able to customize a dashboard, you need the `dashboard/customize` policy.
```

> By default, all the users belonging to the `Editors` User Group, have `Dashboard` role assigned, so they can edit, create, or delete dashboard.

Reviewer: ```suggestion
By default, all users belonging to the `Editors` User Group have the `Dashboard` role assigned, so that they can edit, create, or delete dashboards.
```

Reviewer 2: Dashboard, singular form here is correct.

If, by any reason, you want to narrow this permission, you can set up specific limitations.

For more information about setting up a permission system, see [Work with permissions](work_with_permissions.md).

> !!! Tip
> 
>     For a list of available policies, permissions, and limitations, see [Permissions]([[= developer_doc =]]/permissions/permissions/), [Limitations]([[= developer_doc =]]/permissions/limitations/), and [Policies]([[= developer_doc =]]/permissions/policies/).

Reviewer: ```suggestion
```
I like this tip a lot but it does not belong here. You could add it to the "Work with permissions" article in a separate PR.

# User interface

> ## Dashboard

Reviewer: Since this article belongs to a "Getting started" category and gives brief descriptions of various UI elements, I would strongly suggest leaving a short-ish section here, and creating a subisection in TOC called "Work with dashboard" that would contain two articles: Work with dashboard and Dashboard block reference.

Reviewer 2: I think Dashboard needs a separate section with: Dashboard, Work with dashboard and dashboard blocks.
